File: Data Structures & Algorithms/permutations/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[][]}
     */
    permute(nums) {
        function backtrack(perm, nums, pick) {
            if (perm.length === nums.length) {
                res.push([...perm]);
                return;
            }

            for (let i = 0; i < nums.length; i++) {
                if (!pick[i]) {
                    perm.push(nums[i]);
                    pick[i] = true;
                    backtrack(perm, nums, pick);
                    perm.pop();
                    pick[i] = false;
                }
            }
        }

        let res = [];
        backtrack([], nums, new Array(nums.length).fill(false));
        return res;
    }
}

File: Data Structures & Algorithms/missing-number/submission-2.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        res = len(nums)

        # 這個部分在做的事情就是
        # sum(0~n) - sum(實際有的數字) = 缺少的數字
        for i in range(len(nums)):
            # i 是期望有的數字
            # nums[i] 實際的數字
            res += (i - nums[i])
        
        return res